### DISCUSSION TITLE
Use page.get_text("json", flags=2) to extract the text, and the bbox in the extraction result has a negative number

### DISCUSSION THREAD

**Discussion by zyc130130** [pymupdf/PyMuPDF#1104] [Looking for help] [answered]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
Use page.get_text("json", flags=2) to extract the text, and the bbox in the extraction result has a negative number.

## To Reproduce (mandatory)
        with open(file_path, "rb") as fr:
            file_name = fr.read()
        doc = fitz.open(stream=file_name, filetype="pdf")
        page = doc[page_num-1]
        d2 = page.get_text("json", flags=2)

## Expected behavior (optional)
Describe what you expected to happen (if not obvious).

## Screenshots (optional)
![image](https://user-images.githubusercontent.com/30709006/123028739-7d342000-d412-11eb-8642-6fff1e150d84.png)

## Your configuration (mandatory)
 3.7.6 (default, Jan  8 2020, 20:23:39) [MSC v.1916 64 bit (AMD64)] 
 win32 

PyMuPDF 1.18.14: Python bindings for the MuPDF 1.18.0 library.
Built for Python 3.7 on win32 (64-bit).

For example, the output of `print(sys.version, "\n", sys.platform, "\n", fitz.__doc__)` would be sufficient (for the first two bullets).

## Additional context (optional)
I think the problem is caused by the page margins. After converting the page to a picture, I found that the picture display content has more margins than the PDF page display content.
original PDF page：
![image](https://user-images.githubusercontent.com/30709006/123029289-7a85fa80-d413-11eb-9789-8e63a3d473fd.png)

convert2img：
![image](https://user-images.githubusercontent.com/30709006/123029356-8ec9f780-d413-11eb-819a-33a8a689b907.png)

**Comment by zyc130130** [comment]
I want to get the right height&weight of pdf page(have the page margin, like size of conver2pdf's page), and get the positive bbox values

**Comment by JorjMcKie** [comment, marked as answer]
Negative coordinates are not a bug necessarily. They may happen and if so, it was the PDF creator who is responsible.
By the way: your width and height are **_both positive_**, so this is **_not the problem_**.
If you want to see only those which has positive coordinate, specify a rectangle when extracting text: `page.get_text(..., clip=rect)`.
